p2p/kademlia/dao/key.go

```diff
@@ -0,0 +1,30 @@
+package dao
+
+import (
+	"time"
+)
+
+// Key is the interface for implementing the storage mechanism for the
+// DHT.
+type Key interface {
```

Comment:
Re-read one more time this [request](https://github.com/pastelnetwork/gonode/pull/30#discussion_r637273191)

1. You need to pass context.Context in order to remove `context.Background()`
```go
func Retrieve(db *sql.DB, key []byte) ([]byte, error) {
	var data []byte
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

2. Please make sure your `p2p` package does not create a new context `context.Background()` anywhere.

Reply:
p2p doesn't create context.Background() unless in test files.